### PR TITLE
feat(python): init python component detector and base for inspector for issue #66

### PR DESCRIPTION
**Pull request by ryzzaki**
This PR aims to merge a `PythonComponentDetector` and a base of `PythonPackageInspector` to the `master` branch.

### CODE REVIEW

**Comment by prokopsimek** [comment]
# [DX Scanner](https://dxscanner.io) Report <a href="https://dxscanner.io" target="_blank">DXScanner.io</a>
## https://github.com/DXHeroes/dx-scanner 

<h3>⚠️ Improvements with medium impact
</h3>
<details>
<summary>Solve Issues Continuously</summary>

<b>Do not have an open Issues more than 60 days. Solve Issues continuously.</b> <a href="https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612">https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612</a>
</details>
<details>
<summary>Update Dependencies of Minor and Patch Level</summary>

<b>Keep the dependencies updated to eliminate security concerns and compatibility issues. Use, for example, npm-check-updates.</b> <a href="https://github.com/tjunnone/npm-check-updates">https://github.com/tjunnone/npm-check-updates</a>


 Name                             | New     | Current  
---------------------------------- | --------- | ----------
 @commitlint/lint                 | 8.3.5   | 8.3.4    
 eslint-plugin-eslint-plugin      | 2.2.1   | 2.2.0    
 memfs                            | 3.0.4   | 3.0.3    
 ts-node                          | 8.6.2   | 8.6.1    
 typescript                       | 3.7.5   | 3.7.4    
 @types/node                      | 13.1.8  | 12.12.24 
 semantic-release                 | 16.0.2  | 16.0.1   
 semantic-release-slack-bot       | 1.4.2   | 1.4.1    
 @octokit/rest                    | 16.37.0 | 16.36.0  
 @types/jest                      | 24.9.0  | 24.0.25  
 @typescript-eslint/eslint-plugin | 2.16.0  | 2.15.0   
 @typescript-eslint/parser        | 2.16.0  | 2.15.0   

  <b>Impact changed from high to medium.</b>
</details>
<details>
<summary>ESLint Without Errors</summary>

<b>Use the ESLint correctly. You have some errors.</b> <a href="https://eslint.org/">https://eslint.org/</a>
</details>
<h3>🔔 Improvements with minor impact
</h3>
<details>
<summary>Write Commit Messages by Convention</summary>

<b></b> <a href="https://www.conventionalcommits.org/">https://www.conventionalcommits.org/</a>


 Commit Message                                               | Problems                                                               
-------------------------------------------------------------- | ------------------------------------------------------------------------
 Test/increase codecoverage (#204)                            | subject may not be empty; type may not be empty                        
                                                              |                                                                        
 Test/increase codecoverage                                   |                                                                        
 docs: Travis badge uses a master branch to show build status | subject must not be sentence-case, start-case, pascal-case, upper-case 

</details>
<details>
<summary>Update Dependencies of Major Level</summary>

<b>Keep the dependencies updated to have all possible features. Use, for example, npm-check-updates.</b> <a href="https://github.com/tjunnone/npm-check-updates">https://github.com/tjunnone/npm-check-updates</a>


 Name        | New    | Current  
------------- | -------- | ----------
 @types/node | 13.1.8 | 12.12.24 

</details>
<details>
<summary>Use a JS Logging Library</summary>

<b>Use a logging library to avoid errors and even cyber attacks. The most widely used logging library in the JavaScript community is Winston.</b> <a href="https://www.npmjs.com/package/winston/">https://www.npmjs.com/package/winston/</a>
</details>
🔥 Evaluation of these practices failed
- Break down large pull requests into smaller ones

---



Implementation is not adoption.
We can help you with both. :-)
[dxheroes.io](https://dxheroes.io)



##### Found a bug? Please <a href="https://github.com/DXHeroes/dx-scanner/issues" target="_blank">report</a>.

File: src/inspectors/package/PythonPackageInspector.spec.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    describe('#hasPackage', () => {
+      beforeEach(async () => {
+        await inspector.init();
+      });
+
+      it('Returns true package if it exists', () => {
+        expect(inspector.hasPackage('confluent-kafka')).toBe(true);
+      });
+
+      it('Returns false package if it exists', () => {
```
Comment (adelkahomolova) [inline src/inspectors/package/PythonPackageInspector.spec.ts]:
```suggestion
      it('Returns false if package does not exists', () => {
```

File: src/inspectors/package/PythonPackageInspector.spec.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        expect(pkg).toBeUndefined();
+      });
+    });
+
+    describe('#hasPackage', () => {
+      beforeEach(async () => {
+        await inspector.init();
+      });
+
+      it('Returns true package if it exists', () => {
```
Comment (adelkahomolova) [inline src/inspectors/package/PythonPackageInspector.spec.ts]:
```suggestion
      it('Returns true if package exists', () => {
```

File: src/inspectors/package/PythonPackageInspector.spec.ts (hunk 1 above)
Comment (ryzzaki) [reply src/inspectors/package/PythonPackageInspector.spec.ts]: Thanks! I took the liberty of grabbing `JavaScriptPackageInspector.spec.ts` and fixing the same typos as I have modelled the tests for Python from that source.

**Comment by ryzzaki** [comment]
> @ryzzaki The PR is ok.
> 
> I have just two questions:
> 
> * [ ]  Should we change the 🚧 image for Python to ✅ in readme?
> * [ ]  Should we add a new column `Python` in readme detail for `List of All Practices 🔍` table?

I suppose we could add these changes since DXS is capable of scanning Python projects end to end with Language Independent Practices. I will make these changes in the README in this branch and merge it.

**Comment by prokopsimek** [comment]
Agreed


*Prokop Simek*
prokop.simek@applifting.cz / +420 608 555 937 <+420%20608%20555%20937>

*Applifting s.r.o.*
Šaldova 9 <x-apple-data-detectors://2/1>
Praha 8 <x-apple-data-detectors://2/1>, 186 00
www.applifting.cz <https://applifting.cz>

<https://twitter.com/prokopsim> <https://www.facebook.com/prokopsimek>
<https://www.linkedin.com/in/prokopsimek> <skype:e.s.o.prokop?call>

**Pull request merged**